lib/pipeline.js

```diff
@@ -931,9 +931,18 @@ class PipelineModel extends BaseModel {
                 username,
                 scmContext: this.scmContext
             });
-
-            // eslint-disable-next-line no-await-in-loop
-            const permission = await user.getPermissions(this.scmUri);
+            let permission = {};
+
+            try {
+                // eslint-disable-next-line no-await-in-loop
+                permission = await user.getPermissions(this.scmUri);
+            } catch (err) {
+                if (err.status === 401) {
+                    permission.push = false;
+                } else {
+                    throw err;
```

Comment:
don't throw `err` here let it just continue. Else API will never work if user is invalid, or some other random error.